apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -275,6 +270,14 @@ function main() {
 
     $scope.hocCategoryName = i18n.hoc_category_name;
 
+    // Angular does not offer a reliable way to wait for the template to load,
+    // so do it using a custom event here.
+    $scope.$on('section-page-rendered', () => {
```

Comment:
I'm not sure Angular needs to provide this itself - couldn't you use a [MutationObserver](https://developers.google.com/web/updates/2012/02/Detect-DOM-changes-with-Mutation-Observers) to watch for `#section-page` to be added to the DOM?